src/bundle/ui-dev/src/modules/multi-file-upload/components/upload-popup/upload.popup.component.js

```diff
@@ -1,45 +1,79 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 
+import { getTranslator, getRootNode } from '@ibexa-admin-ui/src/bundle/Resources/public/js/scripts/helpers/context.helper';
+import { parse as parseTooltips } from '@ibexa-admin-ui/src/bundle/Resources/public/js/scripts/helpers/tooltips.helper';
+import { getContentTypeName } from '@ibexa-admin-ui/src/bundle/Resources/public/js/scripts/helpers/content.type.helper';
+
 import TooltipPopup from '../../../common/tooltip-popup/tooltip.popup.component';
 import DropAreaComponent from '../drop-area/drop.area.component';
 import UploadListComponent from '../upload-list/upload.list.component';
 
-const { Translator } = window;
-
 const CLASS_SCROLL_DISABLED = 'ibexa-scroll-disabled';
 
 export default class UploadPopupModule extends Component {
     constructor(props) {
         super(props);
 
         this.refTooltip = React.createRef();
+        this.rootNode = getRootNode();
     }
 
     componentDidMount() {
-        window.document.body.classList.add(CLASS_SCROLL_DISABLED);
-        window.ibexa.helpers.tooltips.parse(this.refTooltip.current);
+        this.rootNode.classList.add(CLASS_SCROLL_DISABLED);
+        parseTooltips(this.refTooltip.current);
     }
 
     componentWillUnmount() {
         window.document.body.classList.remove(CLASS_SCROLL_DISABLED);
     }
 
+    getContentTypesMaxFileSize() {
+        const { locationMappings, defaultMappings, maxFileSize: defaultMaxFileSize } = this.props.adminUiConfig.multiFileUpload;
+        const mappings = locationMappings.length ? locationMappings : defaultMappings;
+        const contentTypeIdentifiers = Object.keys(this.props.contentCreatePermissionsConfig);
+
+        return contentTypeIdentifiers.reduce((maxFileSizes, contentTypeIdentifier) => {
+            const contentTypeName = getContentTypeName(contentTypeIdentifier);
+            const contentTypeMapping = mappings.find((item) => item.contentTypeIdentifier === contentTypeIdentifier);
+
+            maxFileSizes.push({
+                name: contentTypeName,
+                maxFileSize: contentTypeMapping.maxFileSize || defaultMaxFileSize,
```

Comment:
```suggestion
                maxFileSize: contentTypeMapping.maxFileSize ?? defaultMaxFileSize,
```

Reply:
Michał is not always the same thing! maybe max file size is 0?
And we want it to be default if it is 0.